structure_tests/types.go

```diff
@@ -20,7 +20,7 @@ import (
 )
 
 type StructureTest interface {
-	RunAll(t *testing.T)
+	RunAll(t *testing.T) int
```

Comment:
What do you think about adding a method to this interface to contain a length? That way we can error at parse time instead of having to track an int.